client/components/data/purchases/site-purchases/index.jsx

```diff
@@ -1,65 +0,0 @@
-/**
- * External dependencies
- */
-import React from 'react';
-
-/**
- * Internal dependencies
- */
-import { fetchSitePurchases } from 'lib/upgrades/actions';
-import PurchasesStore from 'lib/purchases/store';
-import StoreConnection from 'components/data/store-connection';
-import sitesFactory from 'lib/sites-list';
-
-/**
- * Module variables
- */
-const sites = sitesFactory(),
-	stores = [
-		PurchasesStore,
-		sites
```

Comment:
`stores` is passed to `StoreConnection`, which calls `setState` when any of the stores emit a change event. `connect` will provide us with the similar behavior (through newly provided props) with `purchases`. However, as far as I can tell, when `sites` emits a change event, we no longer rerender `SiteSettings`. This could pose a problem if the user e.g. changes their selected site while the component is mounted.

Potential solutions:

- Pull this data from `state.sites` in `connect`. I'm not sure if this is possible, as I'm not sure if the selected site is available in this branch.
- Use the [data-observe mixin](https://github.com/Automattic/wp-calypso/blob/master/client/lib/mixins/data-observe/index.js) to force the component to render when `sites` emits a change event. This is an anti-pattern, so we should only use it if we need `sites-list` instead of `state.sites`.

Others might be aware of a nicer solution. :)